recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc-caam.sh

```diff
@@ -94,6 +123,25 @@ tdx_enc_partition_mount() {
     fi
 }
 
+# restore data if available
+tdx_enc_restore_data() {
+    if [ ${TDX_ENC_PRESERVE_DATA} -eq 1 ]; then
+        tdx_enc_log "Checking backup file: ${TDX_ENC_PRESERVE_LOCATION}"
+
+        if [ -f ${TDX_ENC_PRESERVE_LOCATION} ]; then
+            tdx_enc_log "Restoring original content..."
+
+            msgs="$(bzip2 -cd ${TDX_ENC_PRESERVE_LOCATION} | tar -C ${TDX_ENC_STORAGE_MOUNTPOINT} -xf -)"
```

Comment:
Not sure what to do here to ensure the errors are raised and captured properly.
See: https://github.com/toradex/meta-toradex-security/pull/27#discussion_r1606716879